presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContains.java

```diff
@@ -78,13 +79,26 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
         Type type = types.get("T");
         TypeSignature valueType = type.getTypeSignature();
         TypeSignature arrayType = parameterizedTypeName(StandardTypes.ARRAY, valueType);
-        MethodHandle methodHandle = methodHandle(ArrayContains.class, "contains", Type.class, MethodHandle.class, Block.class, type.getJavaType());
+
+        MethodHandle methodHandle;
+        if (type.getJavaType() == void.class) {
+            methodHandle = METHOD_HANDLE_UNKNOWN;
+        }
+        else {
+            methodHandle = methodHandle(ArrayContains.class, "contains", Type.class, MethodHandle.class, Block.class, type.getJavaType());
+        }
+
         MethodHandle equalsHandle = functionRegistry.getScalarFunctionImplementation(internalOperator(OperatorType.EQUAL, BooleanType.BOOLEAN, ImmutableList.of(type, type))).getMethodHandle();
         Signature signature = new Signature(FUNCTION_NAME, SCALAR, RETURN_TYPE, arrayType, valueType);
 
         return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle.bindTo(type).bindTo(equalsHandle), isDeterministic(), true, ImmutableList.of(false, false));
     }
 
+    public static Boolean arrayWithUnknownType(Type elementType, MethodHandle equals, Block arrayBlock)
```

Comment:
Instead of omitting the value argument. I would make it `Void` and change the returned `FunctionInfo` to indicate that the last argument is nullable. I was slightly surprised that it works to omit the argument completely